listenstore/listenstore/listenstore.py

```diff
@@ -106,6 +106,13 @@ def insert_postgresql(self, listens):
                     self.log.error(e)
 
     def fetch_listens_from_storage(self, user_id, from_ts, to_ts, limit, order, precision):
+        """ The timestamps are stored as UTC in the postgres datebase while on retrieving
+            the value they are converted to the local server's timezone. So to compare
+            datetime object we need to create a object in the same timezone as the server.
```

Comment:
Timezones are hard!
Just to let you know, I have a gut feeling that this is wrong. I'm going to look into it and get back to you.

Unix timestamps by definition are in UTC, so interpreting it as a local time is the wrong thing to do. Perhaps what we need to do is make sure the datetime object has an explicit timezone. I'll get back to you on this.